tests/svntests.py

```diff
@@ -36,3 +36,212 @@ def test_versionformat_revision(self):
         basename = self.basename(version='foo2')
         th = self.assertTarOnly(basename)
         self.assertTarMemberContains(th, basename + '/a', '2')
+
+    def _check_servicedata(self, expected_dirents=2, revision=2):
```

Comment:
A lot of these tests are copied directly from `gittests.py`.  Please do not copy and paste large chunks of code, as that violates [the DRY rule](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).  Instead, you can share tests between `gittests.py` and `svntests.py` by putting them in a separate class and then using that as a mixin from the `GitTests` and `SvnTests` classes.  However of course it's fine to keep the `svn`-specific tests in this file.